File: src/main/java/org/kite9/diagram/visualization/planarization/rhd/links/ConnectionManager.java
package org.kite9.diagram.visualization.planarization.rhd.links;

import java.util.Collection;

import org.kite9.diagram.common.BiDirectional;
import org.kite9.diagram.model.Connected;
import org.kite9.diagram.visualization.planarization.rhd.GroupPhase.Group;

public interface ConnectionManager extends Collection<BiDirectional<Connected>> {

	public abstract void handleLinks(Group g);

	public boolean hasContradictions();
}

File: src/main/kotlin/org/kite9/diagram/visualization/planarization/rhd/links/ConnectionManager.kt
package org.kite9.diagram.visualization.planarization.rhd.links

import org.kite9.diagram.common.BiDirectional
import org.kite9.diagram.model.Connected
import org.kite9.diagram.visualization.planarization.rhd.GroupPhase

interface ConnectionManager : MutableCollection<BiDirectional<Connected>> {

    fun handleLinks(g: GroupPhase.Group)

    fun hasContradictions(): Boolean

}